### PR TITLE
adds mention parsing for communities

### LINKED ISSUE #99: Add the ability to set a session on a community
As communities come on board they will need the ability to add sessions which are aligned to their community. The task here is to re-eveluate the session/event/community relationship and create a mutation to set a session on that community. 

- [x] Add mention parsing
- [x] Add Community mention adds community slug to session
- [x] Session updates add or remove community slugs on session based on new text. 
- [x] session.communities array, 0 position, is always the event's community slug.

### PR DESCRIPTION
**Pull request by brettski**
closes #99 

Has dependency on thatconference/that-api#34 to be published first.

### CODE REVIEW

File: src/dataSources/cloudFirestore/session.js [hunk 1]
```diff
@@ -88,8 +120,11 @@ function sessions(dbInstance) {
     if (slug) {
       scrubbedSession.slug = slug;
     }
-    dlog('saving session %o', scrubbedSession);
 
+    // mentions
+    await handleMentions({ scrubbedSession, firestore: dbInstance });
```
Comment (csell5) [inline src/dataSources/cloudFirestore/session.js:125]: Total nit, but handleMentions is unlcear as to what it really does.

**Review by csell5 — APPROVED** [review]

**Review by csell5 — APPROVED** [review]
1 nit just on naming.

File: src/dataSources/cloudFirestore/session.js (hunk 1 above)
Comment (brettski) [reply src/dataSources/cloudFirestore/session.js:125]: Something more like, `handleSessionMentions`?

Comment (csell5) [reply src/dataSources/cloudFirestore/session.js:125]: I guess I don't know what `handle` means. It's hard for me not to think of an event handler.

Comment (brettski) [reply src/dataSources/cloudFirestore/session.js:125]: I guess what it is doing is parsing tokens out of text, looking up in slug collection and adding community slugs to session.communities. With that perhaps `parseSessionMentions` work. Thoughts @csell5 ?

Comment (csell5) [reply src/dataSources/cloudFirestore/session.js:125]: parseAndPersistMentions? like I said total nit.

Comment (brettski) [reply src/dataSources/cloudFirestore/session.js:125]: Not sure what persists means here

Comment (brettski) [reply src/dataSources/cloudFirestore/session.js:125]: `parseAndSetMentions`?

**Review by csell5 — APPROVED** [review]

**Pull request merged**